src/commands.rs

```diff
@@ -41,19 +81,51 @@ pub fn client_handshake(i: &[u8]) -> nom::IResult<&[u8], ClientHandshake<'_>> {
         let (i, maxps2) = nom::number::complete::le_u8(i)?;
         let maxps = (maxps2 as u32) << 16 | maxps1 as u32;
         let (i, username) = nom::bytes::complete::take_until(&b"\0"[..])(i)?;
+        let (i, _) = nom::bytes::complete::tag(b"\0")(i)?;
+
+        let (i, auth_response, db) =
+            if capabilities.contains(CapabilityFlags::CLIENT_CONNECT_WITH_DB) {
+                let (i, auth_response) = nom::bytes::complete::tag(b"\0")(i)?;
+                let (i, _) = nom::bytes::complete::tag(b"\0")(i)?;
+
+                let (i, db) = nom::bytes::complete::tag(b"\0")(i)?;
+                let (i, _) = nom::bytes::complete::tag(b"\0")(i)?;
+
+                (i, auth_response, Some(db))
+            } else {
+                (&b""[..], i, None)
+            };
 
         Ok((
             i,
             ClientHandshake {
                 capabilities: CapabilityFlags::from_bits_truncate(cap as u32),
                 maxps,
                 collation: 0,
-                username,
+                username: username.to_vec(),
+                db: db.map(|c| c.to_vec()),
+                auth_response: auth_response.to_vec(),
+                auth_plugin: vec![],
             },
         ))
     }
 }
 
+fn read_length_encoded_number(i: &[u8]) -> nom::IResult<&[u8], u64> {
+    let (i, b) = nom::number::complete::le_u8(i)?;
+    let size: usize = match b {
+        0xfb => return Ok((i, 0)),
+        0xfc => 2,
+        0xfd => 3,
+        0xfe => 8,
+        _ => return Ok((i, b as u64)),
+    };
+    let mut bytes = [0u8; 8];
+    let (i, b) = nom::bytes::complete::take(size)(i)?;
+    bytes[..size].copy_from_slice(b);
+    Ok((i, u64::from_le_bytes(bytes)))
+}
+
```

Comment:
I'm pretty sure we already have this functionality elsewhere, as this isn't the only place we read a lenenc number.